modules/node-shared/src/main/scala/io/constellationnetwork/node/shared/http/p2p/middlewares/MetricsMiddleware.scala

```diff
@@ -0,0 +1,104 @@
+package io.constellationnetwork.node.shared.http.p2p.middlewares
+
+import cats.data.Kleisli
+import cats.effect.kernel.Async
+import cats.syntax.all._
+import eu.timepit.refined.auto._
+import eu.timepit.refined.api.Refined
+import io.constellationnetwork.node.shared.infrastructure.metrics.Metrics
+import org.http4s.HttpRoutes
+import org.http4s.headers.`X-Forwarded-For`
+import scala.concurrent.duration.FiniteDuration
+import java.util.concurrent.TimeUnit
+
+object MetricsMiddleware {
+
+  def apply[F[_]: Async: Metrics](): HttpRoutes[F] => HttpRoutes[F] = { routes =>
+    Kleisli { req =>
+      val startTime = System.nanoTime()
```

Comment:
Should we use `Clock[F]` instead of `System` to grab the current instant?

Reply:
fixed